### ISSUE TITLE
rtx-ruby's ReshimInstaller assumes an interactive session and runs `rtx reshim`, which may result in `No such file or directory - rtx`

### ISSUE THREAD

**Issue by davidkellis** [jdx/rtx#844]
**Describe the bug**

I am using rtx to install ruby non-interactively in an environment in which I have not previously run `eval "$($RTX activate bash)"`.

I have taken the following excerpt from a much larger shell script that I am trying to use to install ruby non-interactively. The abbreviated script is:
```shell
#!/usr/bin/env bash

RTX="$HOME/.local/share/rtx/bin/rtx"
GEM_CMD="$HOME/.local/share/rtx/shims/gem"
if [ -x $RTX ]; then
  if $RUBY_CMD -e "major, minor, patch = RUBY_VERSION.split('.'); exit 1 unless major.to_i >= 3"; then
    echo 'Ruby 3+ is already installed.'
    $GEM_CMD install mygem
    exit 0
  fi
fi
# install a bunch of dependencies...
curl https://rtx.pub/install.sh | sh
$RTX use -g ruby@3.2
$RTX reshim
$GEM_CMD install mygem
```

What happens is that when the script is being evaluated, either attempt to install the gem fails with this:
```
[vagrant@arch ~]$ $GEM_CMD install mygem
ERROR:  While executing gem ... (Errno::ENOENT)
    No such file or directory - rtx
	/home/vagrant/.local/share/rtx/installs/ruby/3.2.2/lib/rubygems_plugin/rubygems_plugin.rb:14:in ``'
	/home/vagrant/.local/share/rtx/installs/ruby/3.2.2/lib/rubygems_plugin/rubygems_plugin.rb:14:in `block in <top (required)>'
	/home/vagrant/.local/share/rtx/installs/ruby/3.2.2/lib/ruby/3.2.0/rubygems/installer.rb:381:in `block in run_post_install_hooks'
	/home/vagrant/.local/share/rtx/installs/ruby/3.2.2/lib/ruby/3.2.0/rubygems/installer.rb:380:in `each'
	/home/vagrant/.local/share/rtx/installs/ruby/3.2.2/lib/ruby/3.2.0/rubygems/installer.rb:380:in `run_post_install_hooks'
	/home/vagrant/.local/share/rtx/installs/ruby/3.2.2/lib/ruby/3.2.0/rubygems/installer.rb:345:in `install'
	/home/vagrant/.local/share/rtx/installs/ruby/3.2.2/lib/ruby/3.2.0/rubygems/resolver/specification.rb:104:in `install'
	/home/vagrant/.local/share/rtx/installs/ruby/3.2.2/lib/ruby/3.2.0/rubygems/request_set.rb:194:in `block in install'
	/home/vagrant/.local/share/rtx/installs/ruby/3.2.2/lib/ruby/3.2.0/rubygems/request_set.rb:182:in `each'
	/home/vagrant/.local/share/rtx/installs/ruby/3.2.2/lib/ruby/3.2.0/rubygems/request_set.rb:182:in `install'
	/home/vagrant/.local/share/rtx/installs/ruby/3.2.2/lib/ruby/3.2.0/rubygems/commands/install_command.rb:214:in `install_gem'
	/home/vagrant/.local/share/rtx/installs/ruby/3.2.2/lib/ruby/3.2.0/rubygems/commands/install_command.rb:230:in `block in install_gems'
	/home/vagrant/.local/share/rtx/installs/ruby/3.2.2/lib/ruby/3.2.0/rubygems/commands/install_command.rb:223:in `each'
	/home/vagrant/.local/share/rtx/installs/ruby/3.2.2/lib/ruby/3.2.0/rubygems/commands/install_command.rb:223:in `install_gems'
	/home/vagrant/.local/share/rtx/installs/ruby/3.2.2/lib/ruby/3.2.0/rubygems/commands/install_command.rb:169:in `execute'
	/home/vagrant/.local/share/rtx/installs/ruby/3.2.2/lib/ruby/3.2.0/rubygems/command.rb:327:in `invoke_with_build_args'
	/home/vagrant/.local/share/rtx/installs/ruby/3.2.2/lib/ruby/3.2.0/rubygems/command_manager.rb:252:in `invoke_command'
	/home/vagrant/.local/share/rtx/installs/ruby/3.2.2/lib/ruby/3.2.0/rubygems/command_manager.rb:192:in `process_args'
	/home/vagrant/.local/share/rtx/installs/ruby/3.2.2/lib/ruby/3.2.0/rubygems/command_manager.rb:150:in `run'
	/home/vagrant/.local/share/rtx/installs/ruby/3.2.2/lib/ruby/3.2.0/rubygems/gem_runner.rb:51:in `run'
	/home/vagrant/.local/share/rtx/installs/ruby/3.2.2/bin/gem:10:in `<main>'
```

The issue is that https://github.com/rtx-plugins/rtx-ruby/blob/main/rubygems-plugin/rubygems_plugin.rb runs `rtx reshim` in a subshell, and in a non-interactive environment, the `rtx` command may not be present, and in my environment, it isn't.

I have worked around this issue for the moment by prepending `export PATH="$HOME/.local/share/rtx/bin:$PATH"` to my script, and then everything works fine, but the real issue is that the ReshimInstaller in https://github.com/rtx-plugins/rtx-ruby/blob/main/rubygems-plugin/rubygems_plugin.rb should run the rtx binary if it is being invoked from within a non-interactive environment, rather than trying to invoke a potentially nonexistent function/alias/etc.

**To Reproduce**

Run the shell script that I've pasted above. Use a real gem reference, perhaps `nokogiri`.

**Expected behavior**

I expect to be able to install a gem in a non-interactive session without having previously run `eval "$($RTX activate bash)"` to add the rtx function to the running environment.

**`rtx doctor` output**

```
[vagrant@arch ~]$ RTX="$HOME/.local/share/rtx/bin/rtx"
[vagrant@arch ~]$ $RTX doctor
rtx version:
  2023.8.7 linux-x64 (892c752 2023-08-24)

build:
  Target: x86_64-unknown-linux-gnu
  Features: DEFAULT, NATIVE_TLS, OPENSSL, SELF_UPDATE
  Built: Thu, 24 Aug 2023 14:57:38 +0000
  Rust Version: rustc 1.71.1 (eb26296b5 2023-08-03)
  Profile: release

shell:
  /bin/bash
  GNU bash, version 5.1.16(1)-release (x86_64-pc-linux-gnu)
  Copyright (C) 2020 Free Software Foundation, Inc.
  License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>

  This is free software; you are free to change and redistribute it.
  There is NO WARRANTY, to the extent permitted by law.

rtx data directory:
  /home/vagrant/.local/share/rtx

rtx environment variables:
  (none)

settings:
  {"always_keep_download": "false", "always_keep_install": "false", "asdf_compat": "false", "disable_default_shorthands": "false", "disable_tools": "[]", "experimental": "false", "jobs": "4", "legacy_version_file": "true", "legacy_version_file_disable_tools": "[]", "log_level": "INFO", "missing_runtime_behavior": "warn", "plugin_autoupdate_last_check_duration": "10080", "raw": "false", "trusted_config_paths": "[]", "verbose": "false", "yes": "false"}

config files:
  /home/vagrant/.config/rtx/config.toml

plugins:
  go       (core)
  java     (core)
  node     (core)
  python   (core)
  ruby     (core)

toolset:
  ruby@3.2

1 problem found:
rtx is not activated, run `rtx activate` for setup instructions
```

**Additional context**

**Comment by jdx** [comment]
I'm not maintaining rtx-ruby anymore in favor of the core ruby plugin. That plugin likely would behave similarly so I added a check if rtx was on PATH. My change is just going to skip running reshim if it can't find rtx. That means if you're using shims you'll need to reshim manually. Perhaps at some point I could look into fixing this but right now I'm not sure how I can reliably find rtx if it's not on PATH. I can't use the absolute path to rtx since it may change.

This is me being slightly pedantic, but the issue isn't around not having an interactive session. The issue is that rtx isn't on PATH. It's normal to use rtx without using `rtx activate`, but using rtx without it being on PATH at all is a more unusual use-case.

**Comment by davidkellis** [comment]
Thank you so much for your help with this @jdx! Thank you also for clarifying the PATH assumptions. I had incorrectly assumed that activate added rtx's bin directory to PATH or added an alias or something, so your explanation helps clarify where I went wrong.

I also didn't realize that I hadn't been using the primary ruby plugin. I had assumed rtx-ruby was the primary/core ruby plugin, and that if I just ran rtx install ruby, then whatever plugin installed ruby would be the primary/default/core plugin.

Admittedly, I haven't read through the entire README, so I'm sure if I would just sit down and read it all the way through that would preempt further mistakes.

In any case, thank you again! rtx is a fantastic tool. I've been using asdf for a while, and rtx fills that niche well, but I'm also really liking that I can use it for ancillary purposes too, like deploying ruby to remote hosts.

Cheers!
